### DISCUSSION TITLE
Save archive to external storage

### DISCUSSION THREAD

**Discussion by sjoerdfirmground** [ArchiveBox/ArchiveBox#940] [Q&A]
Hey everyone, I've been trying to get Archivebox to save to some external store and database rather than the default folder and sqlite db. More specifically, I would like to use DigitalOcean's S3 and managed database. I'm not a python dev myself and I get lost in the code, so I wanted to ask where I should be looking and if there might be plans to integrate this functionality into Archivebox by default?

Cheers

**Comment by pirate** [comment]
This would be a herculean task, I don't think it's worth it. You should check out Archivy or one of the other alternatives in the Wiki instead.

**Reply by caj-larsson** [reply]
~It's definitely not trivial but I think it's doable, the codebase for AB is quite small and I would like it to stay that way.~
~But ,if we assume:~
~* auth* issues will be solved by going through the AB server as a proxy~
~* we only need to support a single storage engine at a time.~

~we should get an MVP by:~

~1. Update the schema to be explicit about content location in a generally addressable way.~
~2. Implement the current storage as a specific implementation of said system~
~3. Implement a data migration between schemas~
~4. Implement an S3 connector~
~5. Build out configuration to set storage engine and give it the necessary data for authorization~

~As for database options, I don't see any reason why we can't lift django's ORM multiple backend support to the user if that is not already done.~

Or we just do what you have already added as an issue https://github.com/ArchiveBox/ArchiveBox/issues/902 :+1:
